### PR TITLE
NIFI-12857 Add test for and refactor implementation of QueuePrioritizer

### PR DESCRIPTION
**Pull request by EndzeitBegins**
# Summary

Adds test for and refactors implementation of `QueuePrioritizer`.

[NIFI-12857](https://issues.apache.org/jira/browse/NIFI-12857)

Additionally, removed some dead code from `TestStandardFlowFileQueue`.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [x] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [x] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Review by exceptionfactory — COMMENTED** [review]
@EndzeitBegins can you rebase this PR from the current main branch? It includes some improvements for the build process that should address the last failure in the Static Analysis workflow step.

**Comment by EndzeitBegins** [comment]
Of course, just rebased on the last commit of `main`.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/queue/QueuePrioritizer.java [hunk 1]
```diff
@@ -18,74 +18,41 @@
...
-                return claimComparison;
-            }
-
-            final int claimOffsetComparison = Long.compare(f1.getContentClaimOffset(), f2.getContentClaimOffset());
-            if (claimOffsetComparison != 0) {
-                return claimOffsetComparison;
-            }
-        }
+        final int claimComparisionResult = claimComparator.compare(f1, f2);
+        if (claimComparisionResult != 0) return claimComparisionResult;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/queue/QueuePrioritizer.java:54]: As a general rule, braces should be used for conditional blocks:
```suggestion
        if (claimComparisionResult != 0) {
            return claimComparisionResult;
        }
```

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/queue/QueuePrioritizer.java [hunk 2]
```diff
@@ -18,74 +18,41 @@
...
-
-        if (f1Penalized && f2Penalized) {
-            if (f1.getPenaltyExpirationMillis() < f2.getPenaltyExpirationMillis()) {
-                return -1;
-            } else if (f1.getPenaltyExpirationMillis() > f2.getPenaltyExpirationMillis()) {
-                return 1;
-            }
-        }
+        final int penaltyComparisonResult = penaltyComparator.compare(f1, f2);
+        if (penaltyComparisonResult != 0) return penaltyComparisonResult;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/queue/QueuePrioritizer.java:46]:
```suggestion
        if (penaltyComparisonResult != 0) {
            return penaltyComparisonResult;
        }
```

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/queue/QueuePrioritizer.java [hunk 3]
```diff
@@ -18,74 +18,41 @@
...
-        if (!prioritizers.isEmpty()) {
-            for (final FlowFilePrioritizer prioritizer : prioritizers) {
-                returnVal = prioritizer.compare(f1, f2);
-                if (returnVal != 0) {
-                    return returnVal;
-                }
-            }
+        for (FlowFilePrioritizer comparator : prioritizers) {
+            final int prioritizerComparisonResult = comparator.compare(f1, f2);
+            if (prioritizerComparisonResult != 0) return prioritizerComparisonResult;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/queue/QueuePrioritizer.java:50]:
```suggestion
            if (prioritizerComparisonResult != 0) {
                return prioritizerComparisonResult;
            }
```

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/queue/QueuePrioritizerTest.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        expiredPenaltyFlowFile.setPenaltyExpiration(System.currentTimeMillis() - 9_001);
+        MockFlowFileRecord penalizedFlowFile = new MockFlowFileRecord(Map.of("penalized", "short"), 0);
+        penalizedFlowFile.setPenaltyExpiration(System.currentTimeMillis() + 123_456);
+        MockFlowFileRecord longerPenalizedFlowFile = new MockFlowFileRecord(Map.of("penalized", "long"), 0);
+        longerPenalizedFlowFile.setPenaltyExpiration(System.currentTimeMillis() + 456_789);
+
+        assertEquals(0, prioritizer.compare(nonPenalizedFlowFile, nonPenalizedFlowFile));
+        assertEquals(1, prioritizer.compare(nonPenalizedFlowFile, expiredPenaltyFlowFile));
+        assertEquals(-1, prioritizer.compare(nonPenalizedFlowFile, penalizedFlowFile));
+        assertEquals(-1, prioritizer.compare(nonPenalizedFlowFile, longerPenalizedFlowFile));        assertEquals(1, prioritizer.compare(longerPenalizedFlowFile, nonPenalizedFlowFile));
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/queue/QueuePrioritizerTest.java:64]:
```suggestion
        assertEquals(-1, prioritizer.compare(nonPenalizedFlowFile, longerPenalizedFlowFile));
        assertEquals(1, prioritizer.compare(longerPenalizedFlowFile, nonPenalizedFlowFile));
```

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
@EndzeitBegins Returning to this PR, I noticed a couple stylistic adjustments needed, but with those and one more rebase, this looks close to completion.

**Comment by EndzeitBegins** [comment]
Thank you for the review @exceptionfactory. 

I've just rebased the PR on the last commit on `main`, added the missing line break to the test, and added braces to conditional blocks where they were missing. 

I would appreciate another look.

**Review by exceptionfactory — APPROVED** [review]
Thanks @EndzeitBegins! +1 merging

**Pull request closed without merging**